### ISSUE TITLE
Allow configuring an element to not be tracked by --track-all

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#354]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/354)
In GitLab by [[Gitlab user @TheRealMichaelCatanzaro]](https://gitlab.com/TheRealMichaelCatanzaro) on Apr 11, 2018, 21:47

This might seem weird, but here is the problem. In https://gitlab.gnome.org/GNOME/gnome-build-meta/merge_requests/18 [[Gitlab user @abderrahimk]](https://gitlab.com/abderrahimk) says that the recommended workflow for GNOME developers using BuildStream should now be:

```
bst track base/freedesktop-sdk-junction.bst
bst track --deps all --except base.bst core/your-module.bst
bst build core/your-module.bst
```

or, a bit simpler:

```
bst track base/freedesktop-sdk-junction.bst
bst build --track-all --track-except base.bst core/your-module.bst
```

But this is really too hard, now. We should try to keep things as simple as possible. Previously, we had a nice workflow where `bst build core/your-module.bst` was equivalent to `jhbuild buildone`, and `bst build --track-all core/your-module.bst` was equivalent to `jhbuild build`. Having to do track separately would be OK, but having to remember extra args like `--deps all --except base.bst` is altogether too hard. Similarly, in the one-line version, `--track-except base.bst` adds too much typing.

One solution would be to somehow configure the project such that base.bst is always skipped when using `--track-all` or `track --deps all`, such that the only way to track it would be to run `bst track base.bst`. Maybe there are other solutions. Thoughts?

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 12, 2018, 05:03

I think the solution to this is to disable cross junction tracking by default, unless a specific option is given.

Besides this, I've been thinking of *"tracking profiles"* for a long time, I'd like to have the project specify some profiles which can be specified by name at track time.

This way one could easily do things like:
* Try the high level stack against refreshed low level dependencies
* Try a new set of testing apps against the version of middleware you're testing
 
But a first step would be to make cross junction tracking disabled by default I think.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 12, 2018, 13:06

mentioned in commit 8d0b421528be94a61b8afdc2488f67d511871219

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 12, 2018, 13:39

mentioned in commit fece8cc81e8d8412e32c6667682a33e7d2f9dafe

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 12, 2018, 13:58

closed via commit fece8cc81e8d8412e32c6667682a33e7d2f9dafe
